data/tutorials/language/0lg_09_env.md

```diff
@@ -0,0 +1,562 @@
+- Write all kinds of definitions, for all kinds of values, in all different ways
+- Accurately assess the scope of any definition
+- Avoid being fooled by shadowed definitions and scopes
+- Explain what a closure is and why it is so named
+- Partially apply functions, create closures, and pass them as values
+- Curry and uncurry functions
+- Use the `unit` type as a function parameter or return value, when needed
+
+### Chatting With OCaml
+
+This tutorial is shown using the UTop toplevel, which can be initiated with the `utop` command. Although other toplevels may be used, we recommend UTop. Code samples are intended to be copied/pasted into UTop. It would benefit the reader to write variations around the examples provided to strengthen understanding. The topics discussed are not limited to interactive execution of OCaml expressions. However, we believe they are easier to understand within the dynamics of interaction with the OCaml interpreter.
+
+### Touch Base on Double Semicolon
+
+When using UTop to interact with the OCaml interpreter, lines ending with double semicolons trigger the parsing, type-checking, and evaluation of everything typed between the prompt and the double semicolon. This may span several lines. However, the interpretation of that double semicolon isn't made by the OCaml interpreter; it is made by UTop, the OCaml toplevel. Once the evaluation of a double semicolon terminated entry is over, the REPL waits for another piece of input.
+
+Nevertheless, the double semicolon `;;` is a valid token in the OCaml syntax. It is a [no-op](https://en.wikipedia.org/wiki/NOP_(code)), i.e., it does not trigger any behaviour, so it is ignored by the compiler. In OCaml source code files meant to be compiled or interpreted as scripts, double semicolons can and should be avoided. Leaving them does not raise errors, but they are useless. The compiler tolerates them to allow copy-paste from UTop to a file without having to remove them.
+
+## Non-Function Values
+
+### Values
+
+Like most functional programming languages, OCaml is an [expression-oriented programming language](https://en.wikipedia.org/wiki/Expression-oriented_programming_language). There are no [statements](https://en.wikipedia.org/wiki/Statement_(computer_science)), i.e., syntactical constructions made to produce some change of state. Variable [assignments](https://en.wikipedia.org/wiki/Assignment_(computer_science)) or [for loop](https://en.wikipedia.org/wiki/For_loop) are examples of statements. Computations triggered by OCaml are written as expressions. Once completed, they produce a value, which has a type. Here are a few examples of expressions, their type, and resulting values.
```

Comment:
I would phrase the difference between expressions and statements this way: the evaluation of a an expression yields a value, the evaluation of a statement yields an effect, i.e. a change of state. In many languages expressions can still yield effects and so the distinction between is not always clear cut. The same is true in OCaml where expressions and effects can be mixed.